src/filters/dec_scte35.c

```diff
@@ -470,6 +473,15 @@ static GF_Err scte35dec_process_emsg(SCTE35DecCtx *ctx, const GF_PropertyValue *
 	return GF_OK;
 }
 
+static Bool scte35dec_is_splice_point(SCTE35DecCtx *ctx, u64 dts)
+{
+	Event *evt = gf_list_get(ctx->ordered_events, 0);
+	if (!evt) return GF_FALSE;
+	Bool is_splice = (evt->dts + evt->emib->presentation_time_delta == dts);
```

Comment:
This compares DTS and PTS. Luckily pts=dts for most IDRs but not all (cf PTS-offset put by Elemental in their HLS streams).

Also not all events trigger an IDR, we need investigate that.

Reply:
This comparison didn't feel natural to me as well. Honestly I set this to CTS initially but within this filter all comparisons were made with DTS so I kept it same.

https://github.com/gpac/gpac/blob/21284f79f61119bbc8dbe7669a832b71b754a90a/src/filters/dec_scte35.c#L185

Reply:
dts <= pts (because you need to decode before being able to present). That's why it seems an ok estimation when deciding to flush the box. But it is not when pushing an IDR on an exact frame.

We should certainly consider getting the cts/pts as well and add some unit tests.

Reply:
Gotcha, I'll revert to my initial version with CTS then

Reply:
Also this presentation time offset is an offset from the DTS of this packet right? Or is it it's CTS? I may have missed this information on the specs

Reply:
I've modified the code so that only the following conditions instruct an IDR:

- `splice_insert` and `splice_event_cancel_indicator == 0`
- `time_signal` and `segmentation_type_id == 0x10`
- `time_signal` and `segmentation_type_id == 0x34`